### ISSUE TITLE
As a user, I'd like an API or csv export of metrics to analyze them and import them into dashboards

### ISSUE THREAD

**Issue by abinoda** [pullreminders/backlog#156]
From @kompio:

> Hi Abi, Can I export pull reminder's data ? or Would you plan to provide API for us?We'd like to analyze pull reminder's data on our dashboard service.

From @LeZuse:

> I wanted to get a number badge (like some kind of Mac bar menu app or similar) that would tell me there are followups to pay attention to. Your product seemed like the good place to get that information, but I didn't find any API for it.

Also requested by @willyyang, @julman99.

**Comment by willyyang** [comment]
I think I have the same use case as @kompio.. I want to fetch analytics / analyze our pull reminder data in order to create dashboards for my internal teams. 

If we're able to export iframes of the analytics charts that would be nice too..

**Comment by abinoda** [comment]
@willyyang What tool would you use to create dashboards?

**Comment by willyyang** [comment]
it's built in house.

**Comment by joscha** [comment]
Our organization would also be interested in such an API. We use the Chromium owners model for code ownership and as such the people who have to review a pull request are calculated from this ownership structure. We currently have a tool that comments on a pull request and tells you the people you need to assign. It includes the load (how many pull requests waiting for review on that particular person) but not how long it is most likely going to take the person to come round to your pull request if you were to assign them. In order to facilitate a more even load between reviewers in the case there is a choice of reviewers I'd like to show an indication of the median turnaround time for that person at the given point in time. That data is available within pull panda.

**Comment by kompiro** [comment]
@abinoda 

Hi abi, Congratulation to join to GitHub. I really glad to the big news!
I hope pullpanda provides some api as GitHub API.

**Comment by willyyang** [comment]
@abinoda Congrats! Any news on this issue?

**Comment by abinoda** [comment]
@kompiro @willyyang Thanks guys! I'm actually looking into this right now. Could you tell me more specifics on both of your use cases? Like, what exact dashboards are you interested in building in your internal dashboard tools?

**Comment by willyyang** [comment]
Just looking for API that backs the data / graphs generated in the `Analytics` section of PullReminders. 

Best case is if you're able to allow for exporting iframes of the analytics charts.. If not, I just need the data and I can re-create the graphs.

**Comment by abinoda** [comment]
@willyyang if you had the data would you create different graphs than what is shown in the app? Or would you recreate the same ones.

**Comment by willyyang** [comment]
I'd re-create the same ones. You've done a great job showcasing basically everything my team and I want!

**Comment by tristanbes** [comment]
Hello 🌝,

Let me add a use case, I was hoping to display some metrics on a dashboard:
Behold for the beautiful mockup.


![Esquisse](https://user-images.githubusercontent.com/346010/60529898-e6876980-9cf7-11e9-8ed4-21598996a9b6.png)


Note that if this dashboard exist inside the pullreminders.com i'd be happy to display it on the TV instead of consuming an API (Accessing the URL it via a private token for example, to avoid logging the TV with a Github account -for obvious security reasons-).

**Comment by pelzhaus** [comment]
Agreed with all of the above! Another example that would be nice is to periodically pull data to create our own time series graphs - there are some in the analytics section but I find them somewhat limited (I _personally_ dont love stacked bar charts, but to each their own), and would love to display them next to arbitrary data (key metrics, tickets, etc.). Some questions I'd like to answer over time:

- Over the course of a day what time do PRs usually come in? What time do reviews come in?
- What's the review breakdown of PR into different size buckets? (e.g. do small PRs get approved quicker or have more comments)
- What % of PRs get approved on the first time? What % have multiple "request changes"? Can we group that by team or developer?
- How quickly to PRs with certain labels (e.g. Urgent / Bug) get reviewed relative to others (e.g. Enhancement)?

Being able to sample the raw data via CSV or API at a regular interval to make this possible would be great!

**Comment by abinoda** [comment]
@tristanbes What is "success rate" in your illustration?

**Comment by abinoda** [comment]
@pelzhaus For my own understanding, could you elaborate more on the use case for why you'd like to understand these:

> Over the course of a day what time do PRs usually come in? What time do reviews come in?

> What % of PRs get approved on the first time? What % have multiple "request changes"? Can we group that by team or developer?

**Comment by willyyang** [comment]
@abinoda @pelzhaus I think this allows for better measurement of team velocity and just exposes more metrics that can be used to refine how teams work.

**Comment by pelzhaus** [comment]
@willyyang exactly, just working through different metrics that may help expose patterns of the team. 

An example for the time of day stat: Lets say you do a daily deployment that starts at noon every day. Does the team exclusively submit pull requests at 11:30 and try to get them into deployment that day? Do people tend to submit reviews first thing in the morning or later in the day? Exposing these habits to the team would help us be more thoughtful about how we work.

**Comment by jlisee** [comment]
Any update on the data export?  One thing our team would like to measure is how different workflows tools affect the size of a pull request and how many comments they get.  It would be awesome if we could build of the work done by pullreminders here vs. setting up own system.
